examples/hal/quad/main.rs

```diff
@@ -499,62 +349,186 @@ fn main() {
                     image_layers: i::SubresourceLayers {
                         aspects: f::Aspects::COLOR,
                         level: 0,
-                        layers: 0 .. 1,
+                        layers: 0..1,
                     },
                     image_offset: i::Offset { x: 0, y: 0, z: 0 },
-                    image_extent: i::Extent { width, height, depth: 1 },
-                }]);
+                    image_extent: i::Extent {
+                        width,
+                        height,
+                        depth: 1,
+                    },
+                }],
+            );
 
             let image_barrier = m::Barrier::Image {
-                states: (i::Access::TRANSFER_WRITE, i::Layout::TransferDstOptimal) ..
-                        (i::Access::SHADER_READ, i::Layout::ShaderReadOnlyOptimal),
+                states: (i::Access::TRANSFER_WRITE, i::Layout::TransferDstOptimal)
+                    ..(i::Access::SHADER_READ, i::Layout::ShaderReadOnlyOptimal),
                 target: &image_logo,
                 range: COLOR_RANGE.clone(),
             };
             cmd_buffer.pipeline_barrier(
-                PipelineStage::TRANSFER .. PipelineStage::FRAGMENT_SHADER,
+                PipelineStage::TRANSFER..PipelineStage::FRAGMENT_SHADER,
                 m::Dependencies::empty(),
                 &[image_barrier],
             );
 
             cmd_buffer.finish()
         };
 
-        let submission = Submission::new()
-            .submit(Some(submit));
-        queue.submit(submission, Some(&mut frame_fence));
+        let submission = Submission::new().submit(Some(submit));
+        queue_group.queues[0].submit(submission, Some(&mut frame_fence));
 
         device.wait_for_fence(&frame_fence, !0);
     }
 
+    #[cfg(feature = "vulkan")]
```

Comment:
Only vulkan requires recreating the swapchain, the rest only need updating the viewport. It would be a waste of performance if we recreated it in other backends.